Clang complains about unused attribute in `qemu_virtual_machine.h`

```
In file included from /home/ricab/mydev/multipass/multipass/src/platform/backends/qemu/qemu_virtual_machine.cpp:18: 
/home/ricab/mydev/multipass/multipass/src/platform/backends/qemu/qemu_virtual_machine.h:67:27: error:
 private field 'process_factory' is not used [-Werror,-Wunused-private-field] 
    const ProcessFactory* process_factory; 
                          ^ 
1 error generated. 
```

--- a/src/platform/backends/qemu/qemu_virtual_machine.cpp
+++ b/src/platform/backends/qemu/qemu_virtual_machine.cpp
@@ -158,7 +158,6 @@ mp::QemuVirtualMachine::QemuVirtualMachine(const ProcessFactory* process_factory
       tap_device_name{tap_device_name},
       mac_addr{desc.mac_addr},
       username{desc.ssh_username},
-      process_factory{process_factory},
       dnsmasq_server{&dnsmasq_server},
       monitor{&monitor},
       vm_process{make_qemu_process(process_factory, desc, tap_device_name, mac_addr)},

--- a/src/platform/backends/qemu/qemu_virtual_machine.h
+++ b/src/platform/backends/qemu/qemu_virtual_machine.h
@@ -64,7 +64,6 @@ private:
     const std::string tap_device_name;
     const std::string mac_addr;
     const std::string username;
-    const ProcessFactory* process_factory;
     DNSMasqServer* dnsmasq_server;
     VMStatusMonitor* monitor;
     std::unique_ptr<QProcess> vm_process;

